### PR TITLE
Remove job from commit

### PR DESCRIPTION
**Pull request by runcom**
/cc @LK4D4 My attempt to remove job from commit but then it still need job from builder, hope this is ok to continue with.

Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

**Comment by LK4D4** [comment]
@runcom I think it is first engine-removal which adds more lines, than remove :)

**Comment by runcom** [comment]
@LK4D4 you're right but I had to adapt everything from that env.*

**Comment by runcom** [comment]
I'm trying to figure out why some tests failed btw

File: daemon/commit.go [hunk 1]
```diff
@@ -27,29 +31,34 @@ func (daemon *Daemon) ContainerCommit(job *engine.Job) error {
...
 
+	changesJson, err := json.Marshal(c.Changes)
+	if err != nil {
+		return "", err
+	}
+
 	buildConfigJob := daemon.eng.Job("build_config")
 	buildConfigJob.Stdout.Add(stdoutBuffer)
-	buildConfigJob.Setenv("changes", job.Getenv("changes"))
+	buildConfigJob.Setenv("changes", string(changesJson))
```
Comment (LK4D4) [inline daemon/commit.go]: I think you can use `SetenvJson(c.Changes)`

File: daemon/commit.go [hunk 2]
```diff
@@ -3,22 +3,26 @@ package daemon
...
-	}
-	name := job.Args[0]
+type ContainerCommitConfig struct {
+	Pause   bool
+	Repo    string
+	Tag     string
+	Author  string
+	Comment string
+	Changes []string
+	Config  *io.ReadCloser
```
Comment (LK4D4) [inline daemon/commit.go]: I believe that `io.ReadCloser` is interface, so pointer to it makes no sense.

File: api/server/server.go [hunk 3]
```diff
@@ -618,39 +618,37 @@ func postCommit(eng *engine.Engine, version version.Version, w http.ResponseWrit
...
-	job.SetenvList("changes", r.Form["changes"])
-	job.SetenvSubEnv("config", &config)
+	containerCommitConfig := &daemon.ContainerCommitConfig{
+		Pause:   pause,
+		Repo:    r.Form.Get("repo"),
+		Tag:     r.Form.Get("tag"),
+		Author:  r.Form.Get("author"),
+		Comment: r.Form.Get("comment"),
+		Changes: r.Form["changes"],
+		Config:  &r.Body,
```
Comment (LK4D4) [inline api/server/server.go]: r.Body implements `io.ReadCloser`, no need to get address.

**Comment by runcom** [comment]
@LK4D4 tests would be ok now, basically I've removed the first dependency from commit on engine/job but yet, daemon commit still has a dependency on engine/job with build_config job from builder (discussed on IRC)

**Comment by runcom** [comment]
Removed integration/server_test.go tests because of https://github.com/docker/docker/pull/12282

**Comment by LK4D4** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**